### PR TITLE
work around julia issue #24650 by converting position to Int

### PR DESCRIPTION
**Pull request by KristofferC**
Ref: https://github.com/JuliaLang/julia/issues/24650

### CODE REVIEW

**Comment by wildart** [comment]
Could you add x86 system to travis config to test it?

**Comment by KristofferC** [comment]
It is tested on Appveyor, just need to add a test for an `IOStream`. Will do that.

File: src/parser.jl [hunk 1]
```diff
@@ -50,15 +50,15 @@ end
...
+Base.position(p::Parser) = Int(position(p.input))+1
 Base.read(p::Parser) = read(p.input, Char)
 
 "Rewind parser input on `n` characters."
 function rewind(p::Parser, n=1)
     pos = position(p.input)
     pos == 0 && return 0
     skip(p.input, -n)
-    return position(p.input)
+    return Int(position(p.input))
```
Comment (stevengj) [inline src/parser.jl:61]: Why is this a correct change?  If `p.input` is a large file on a 32-bit machine, won't this fail for files > 2GB?

**Comment by stevengj** [comment]
What problem is this PR solving?

**Comment by stevengj** [comment]
If you're worried about type stability, the right thing seems like it should be to parameterize `Parser` on the IO type, i.e.

```jl
mutable struct Parser{T<:IO}
    input::T
    ...
end
```

**Comment by KristofferC** [comment]
> What problem is this PR solving?

Failure to work at all on 32 bit (when reading files), because function arguments in this package are typed on `::Int`. I did it this way since it was the least intrusive change and I wasn't too worried about 2 GB TOML files.

**Comment by stevengj** [comment]
Seems like it would be better to remove the overtyping of arguments.  Can't you just do a search and replace of `::Int` with `::Integer`?

**Pull request closed without merging**